docs/CHANGELOG/CHANGELOG-1.12.md

```diff
@@ -33,6 +33,196 @@
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+# v1.12.0
+## Downloads for v1.12.0
+
+Download v1.12.0 in the [v1.12.0 release page](https://github.com/karmada-io/karmada/releases/tag/v1.12.0).
+
+## What's New
+
+### OverridePolicy now support partially override values inside JSON and YAML fields
+
+This release introduces a new feature for `OverridePolicy` that allows users to partially override specific values in JSON and YAML resources, rather than replacing the entire configuration. 
+This enhancement ensures minimal modifications and improves ease of use, catering to scenarios where users only want to adjust certain values.
+
+The allowed operations are as follows:
+
++ `add`: appends new key-value pairs at the specified sub path.
++ `remove`: removes specific key-value pairs at the specified sub path.
++ `replace`: replaces existing values with new values at the specified sub path.
+
+For a detailed description of this feature, see the [Proposal: Structured configuration overrider](https://github.com/karmada-io/karmada/tree/master/docs/proposals/structured-configuration)
+
+(Feature contributors: @Patrick0308, @sophiefeifeifeiya, @chaunceyjiang)
+
+### Karmada Operator Enhancement: Multi-Data Center Redundancy and Disaster Recovery(DR)
+
+This release, karmada-operator introduces support for a Custom CA Certificate for Karmada Instances and the ability to utilize external etcd. 
+These enhancements allow Karmada-operator to deploy a highly available managed Karmada control plane across multiple management clusters that can span various data centers, thus fulfilling disaster recovery requirements.
+
+By implementing this architecture and configuring the managed control plane instances to use the same CA certificate and underlying etcd instance, you can create a stretched instance that operates across multiple management clusters. 
+This setup allows secure access through a unified, load-balanced API endpoint. Ultimately, this arrangement enhances resilience against data center outages, complies with disaster recovery requirements, and minimizes the risk of service disruptions.
+
+For a detailed description of this feature, see the [Proposal: Support Custom CA Certificate for Karmada Control Plane](https://github.com/karmada-io/karmada/tree/master/docs/proposals/karmada-operator/custom_ca_cert)
+
+(Feature contributors: @jabellard)
+
+### Karmdactl now supports autocompletion
+
+In the daily operation and maintenance of the Karmada system, many Karmadactl commands are frequently used. 
+
+With this release, Karmadactl now supports autocompletion, which reduces typing significantly and enhances both efficiency and user experience in operating and maintaining Karmadactl. 
+
+Currently, Karmadactl supports command completion for both Bash and Zsh. More information about `karmadactl completion` can be obtained using `karmadactl completion --help`.
+
+(Feature contributors: @zhzhuang-zju)
+
```

Comment:
Add two more new feature describetion:

### Support Roll Back Migration Safely

This release introduced `PreserveResourcesOnDeletion` field to both PropagationPolicy and ClusterPropagationPolicy API, which provides the ability to rollback migration safely. 

The `PreserveResourcesOnDeletion` filed controls whether resources should be preserved on the member clusters when the resource template is deleted in the Karmada control-plane. If it set to true, resources will be preserved on the member clusters.

This feature is particularly useful during workload migration scenarios to ensure that rollback can occur quickly without affecting the workloads running on the member clusters.

For a detailed description of this feature, please refer to [How to Roll Back Migration Operations](https://karmada.io/docs/next/administrator/migration/migrate-in-batch/#how-to-roll-back-migration-operations) and [docs(proposal): Migration Rollback Protection](https://github.com/karmada-io/karmada/pull/5101). 

(Feature contributors: @CharlesQQ @XiShanYongYe-Chang @RainbowMango @a7i @wulemao)

### Stateful Application Failover Supprt

This release introduces a new feature for stateful application failover, it provides a generalized way for users to define application state preservation in the context of cluster-to-cluster failovers.

In the previous releases, Karmada’s scheduling logic runs on the assumption that resources that are scheduled and rescheduled are stateless. In some cases, users may desire to conserve a certain state so that applications can resume from where they left off in the previous cluster.

For CRDs dealing with data-processing (such as Flink or Spark), it can be particularly useful to restart applications from a previous checkpoint. That way applications can seamlessly resume processing data while avoiding double processing.

For a detailed description of this feature, please refer to [[Feature] Stateful Application Failover Support](https://github.com/karmada-io/karmada/issues/5788).

(Feature contributors: @Dyex719 @mszacillo @RainbowMango @XiShanYongYe-Chang)